api/src/middleware/cloudflare-filter/cloudflare-filter.middleware.ts

```diff
@@ -0,0 +1,29 @@
+import { Injectable, NestMiddleware, HttpException, HttpStatus } from '@nestjs/common';
+import { Request, Response } from 'express';
+import { CloudflareIpService } from '../../services/cloudflareip/cloudflare-ip.service';
+import { inRange } from 'range_check';
+
+@Injectable()
+export class CloudflareFilterMiddleware implements NestMiddleware {
+  constructor(private cloudflareIpService: CloudflareIpService) {}
+
+  use(req: Request, res: Response, next: (any?) => void) {
+    if (process.env.ENVIRONMENT !== 'dev') {
+      const ipType = req.ip.match(':') ? 6 : 4;
+      this.cloudflareIpService.getIPs(ipType).toPromise().then(
```

Comment:
IMO its not necessary to query this on every request as it will introduce additional latency and potential for unexpected failures.

This list is essentially static and given most users use iptables any changes to this list will probably occur way in advance of actually being used. Perhaps we lookup this list at startup or build time and cache it.